bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/AddonInfoRegistry.java

```diff
@@ -62,14 +66,60 @@ protected void removeAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
     /**
      * Returns the add-on information for the specified add-on ID and locale (language),
      * or {@code null} if no add-on information could be found.
+     * <p>
+     * If more than one provider provides information for the specified add-on ID and locale,
+     * it returns merged information from all such providers.
      *
-     * @param id the ID to be looked for
+     * @param targetId the ID to be looked for
      * @param locale the locale to be used for the add-on information (could be null)
      * @return a localized add-on information object (could be null)
      */
-    public @Nullable AddonInfo getAddonInfo(String id, @Nullable Locale locale) {
-        return addonInfoProviders.stream().map(p -> p.getAddonInfo(id, locale)).filter(Objects::nonNull).findAny()
-                .orElse(null);
+    public @Nullable AddonInfo getAddonInfo(String targetId, @Nullable Locale locale) {
+        // note: using funky code to prevent a maven compiler error
+        Stream<AddonInfo> addonInfos = addonInfoProviders.stream()
+                .map(p -> Optional.ofNullable(p.getAddonInfo(targetId, locale))).filter(o -> o.isPresent())
+                .map(o -> o.get());
+
+        // one or zero entries
+        if (addonInfos.count() <= 1) {
```

Comment:
count is a terminal operator, so the next statement will give an error. You can convert to list in line 81 already and check the size of the list here.

Reply:
Ok. Thanks for the tip.